satpy/composites/__init__.py

```diff
@@ -413,13 +423,21 @@ def __call__(self, projectables, optional_datasets=None, **info):
             lons, lats = nir.info["area"].get_lonlats()
             sun_zenith = sza(nir.info['start_time'], lons, lats)
 
-        refl39 = Calculator(nir.info['platform_name'],
-                            nir.info['sensor'], nir.id.wavelength[1])
+        self._refl3x = Calculator(nir.info['platform_name'], nir.info['sensor'], nir.id.name)
```

Comment:
Maybe this should be a separate stage prior to calling  `_get_reflectance()`? My suggestion is to add `_init_refl3x()` that does everything else but doesn't return the reflectance, and `_get_reflectance()` would have a check `if self._refl3x is None: self._refl3x()` before the return.

Reply:
Yeah, I see what you mean, but isn't it a bit overkill here?
I think the amount of code would increase actually, and it is not likely this code will be reused elsewhere. And I think you then need even more new attributes besides _refl3x, to hold the nir, and tb11 for instance...

Reply:
The amount of code would increase a little, but the usage on line 437 wouldn't rely on a side-effect any more, and it would be much clearer what is happening.

Lines 407 to 424 would go with the `return` line to `_get_reflectance()`, and lines 400 to 406 + 426 would go to `_init_refl3x()`. (Line 406 would be in both.)